src/core/decoder.js

```diff
@@ -139,10 +118,19 @@ function getLineNormal(a, b) {
     return normalize([-dy, dx]);
 }
 
+function getCornerVector(a, b, c) {
```

Comment:
This returns the normal for all cases where the line segment is not the beginning nor the end of the line string, right?

If that is the case I would change the name, `cornerVector` sounds to me like beginning/end. I would say something like `getJointNormal`.